.github/workflows/provenance.yml

```diff
@@ -0,0 +1,152 @@
+# Copyright The GOSST team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA provenance generator
+
+permissions:
+  contents: read
+
+env:
+  # This repo.
+  GENERATOR_REPOSITORY: slsa-framework/slsa-github-generator
+  GENERATOR_REF: main
+
+###################################################################
+#                                                                 #
+#            Input and output argument definitions                #
+#                                                                 #
+###################################################################
+on:
+  workflow_call:
+    inputs:
+      artifacts:
+        description: "A comma delimited list of artifacts of the form NAME@SHA256"
+        required: true
+        type: string
+    outputs:
+      attestation-name:
+        description: "The artifact name of the signed provenance"
+        value: ${{ jobs.sign-prov.outputs.prov-name }}
+
+jobs:
+  ###################################################################
+  #                                                                 #
+  #                       Build the generator                       #
+  #                                                                 #
+  ###################################################################
+  generator:
+    outputs:
+      go-generator-sha256: ${{ steps.generator-gen.outputs.go-generator-sha256 }}
+      go-prov-sha256: ${{ steps.generator-gen.outputs.go-prov-sha256 }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.3.4
+        with:
+          fetch-depth: 0
+          repository: "${{ env.GENERATOR_REPOSITORY }}"
+          ref: "${{ env.GENERATOR_REF }}"
```

Comment:
Shouldn't this check out the exact version from the reusable workflow? For example, if someone calls `uses: slsa-framework/slsa-github-generator@abcd1234...`, shouldn't this check out `abcd1234...`?

But sadly I can't find a way to get it via the context. Any ideas?

Reply:
I don't think it's available in the context and we need an OIDC request to get it, I think (That's how we're setting the `buildID` in the Go generator)

Reply:
Right, it's not in the context, but I do think it's an important problem to solve. For one, always checking out at head means that we can't properly test it without submitting.

Could a prior step pull out the repository and ref from the OIDC token?

Reply:
We should use OIDC in a different job, and use the results as input to the checkout - since workflow steps are not isolated from each others

Reply:
Right, the intention was never to keep this as it is forever. Using the OIDC token gives the ref that the caller workflow used in their reference to the reusable workflow yaml I suppose?

Reply:
correct

Reply:
I plan to do this in a follow up PR. Issue #4 will track this.